litellm/proxy/db/redis_update_buffer.py

```diff
@@ -0,0 +1,194 @@
+        """
+        if self.redis_cache is None:
+            return None
+        user_transaction_keys = await self.redis_cache.async_scan_iter(
+            "user_list_transactions:*"
+        )
+        end_user_transaction_keys = await self.redis_cache.async_scan_iter(
+            "end_user_list_transactions:*"
+        )
+        key_transaction_keys = await self.redis_cache.async_scan_iter(
+            "key_list_transactions:*"
+        )
+        team_transaction_keys = await self.redis_cache.async_scan_iter(
+            "team_list_transactions:*"
+        )
+        team_member_transaction_keys = await self.redis_cache.async_scan_iter(
+            "team_member_list_transactions:*"
+        )
+        org_transaction_keys = await self.redis_cache.async_scan_iter(
+            "org_list_transactions:*"
+        )
+
+        user_list_transactions = await self.redis_cache.async_batch_get_cache(
+            user_transaction_keys
+        )
+        end_user_list_transactions = await self.redis_cache.async_batch_get_cache(
+            end_user_transaction_keys
+        )
+        key_list_transactions = await self.redis_cache.async_batch_get_cache(
+            key_transaction_keys
+        )
+        team_list_transactions = await self.redis_cache.async_batch_get_cache(
+            team_transaction_keys
+        )
+        team_member_list_transactions = await self.redis_cache.async_batch_get_cache(
+            team_member_transaction_keys
+        )
+        org_list_transactions = await self.redis_cache.async_batch_get_cache(
+            org_transaction_keys
+        )
+
+        # filter out the "prefix" from the keys using the helper method
+        user_list_transactions = self._remove_prefix_from_keys(
+            user_list_transactions, "user_list_transactions:"
+        )
+        end_user_list_transactions = self._remove_prefix_from_keys(
+            end_user_list_transactions, "end_user_list_transactions:"
+        )
+        key_list_transactions = self._remove_prefix_from_keys(
+            key_list_transactions, "key_list_transactions:"
+        )
+        team_list_transactions = self._remove_prefix_from_keys(
+            team_list_transactions, "team_list_transactions:"
+        )
+        team_member_list_transactions = self._remove_prefix_from_keys(
+            team_member_list_transactions, "team_member_list_transactions:"
+        )
+        org_list_transactions = self._remove_prefix_from_keys(
+            org_list_transactions, "org_list_transactions:"
+        )
+
+        return DBSpendUpdateTransactions(
+            user_list_transactions=user_list_transactions,
+            end_user_list_transactions=end_user_list_transactions,
+            key_list_transactions=key_list_transactions,
+            team_list_transactions=team_list_transactions,
+            team_member_list_transactions=team_member_list_transactions,
+            org_list_transactions=org_list_transactions,
+        )
```

Comment:
TODO: Expose a helper to clear out the updates from Redis once they are committed to the DB